### PR TITLE
Add plus minus buttons

### PR DESCRIPTION
**Pull request by sergeymaksimenko**

### CODE REVIEW

File: js/script.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+window.onload = function () {
```
Comment (kozubenko) [inline js/script.js:1]: rm space

File: css/styles.css [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  width: 50px;
+  height: 50px;
+  background-color: rgb(176, 1, 0);
+  margin-left: 51px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
+.buttonMinusRow {
```
Comment (kozubenko) [inline css/styles.css]: rename

File: css/styles.css [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  background-color: rgb(240, 167, 0);
+  margin-top: 1px;
+  margin-left: 51px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
+.buttonMinusColumn {
+  width: 50px;
```
Comment (kozubenko) [inline css/styles.css]: rm copy past, add common classes for buttons

File: css/styles.css [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  height: 100vh;
+  align-items: center;
+  justify-content: center;
+}
+
+.container {
+  max-height: max-content;
+}
+
+.buttonPlusColumn {
```
Comment (kozubenko) [inline css/styles.css]: move down

File: index.html [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>Coubs</title>
+  <link rel="stylesheet" href="css/styles.css">
+  <script src="js/script.js" defer></script>
+</head>
+<body>
+<div class="layout">
+  <div class="container" onmouseover="hideButtons()">
```
Comment (kozubenko) [inline index.html]: `class="container"` check, maybe do not need

File: js/script.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    let cellElements = document.getElementsByClassName('cell');
```
Comment (kozubenko) [inline js/script.js]: const

File: js/script.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    Array.prototype.forEach.call(cellElements, function (element) {
+        element.addEventListener('mousemove', function () {
```
Comment (kozubenko) [inline js/script.js]: rm space
`mousemove` - check maybe mouse over is better

File: js/script.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
+window.onload = function () {
+    let cellElements = document.getElementsByClassName('cell');
+    Array.prototype.forEach.call(cellElements, function (element) {
```
Comment (kozubenko) [inline js/script.js]: `cellElements.forEach((element) => ...)`

File: js/script.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+function hideButtons(event) {
+    if (!event.relatedTarget.classList.contains('cell')) {
+        let buttonMinusColumn = document.querySelector('.buttonMinusColumn');
+        let buttonMinusRow = document.querySelector('.buttonMinusRow');
+        buttonMinusColumn.style.visibility = 'hidden';
+        buttonMinusRow.style.visibility = 'hidden';
+    }
+}
```
Comment (kozubenko) [inline js/script.js]: add line

File: js/script.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    let buttonMinusRow = document.querySelector('.buttonMinusRow');
+    buttonMinusColumn.style.visibility = 'visible';
+    buttonMinusRow.style.visibility = 'visible';
+    let table = document.querySelector('.table');
+    buttonMinusColumn.style.marginLeft = event.pageX - table.offsetLeft + (element.offsetWidth / 2) + 'px';
+    buttonMinusRow.style.marginTop = event.pageY - table.offsetTop - (element.offsetHeight / 2) + 'px';
+}
+
+function hideButtons(event) {
+    if (!event.relatedTarget.classList.contains('cell')) {
```
Comment (kozubenko) [inline js/script.js]: хуйово

File: js/script.js [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+    Array.prototype.forEach.call(cellElements, function (element) {
+        element.addEventListener('mousemove', function () {
+            moveButtons(element)
+        }, false);
+        element.addEventListener('mouseout', hideButtons, false);
+    });
+};
+
+function moveButtons(element) {
+    let buttonMinusColumn = document.querySelector('.buttonMinusColumn');
```
Comment (kozubenko) [inline js/script.js]: play from table

File: js/script.js (hunk 8 above)
Comment (sergeymaksimenko) [reply js/script.js]: не получилось (

File: js/script.js (hunk 10 above)
Comment (sergeymaksimenko) [reply js/script.js]: решил иначе

File: js/script.js [hunk 12]
```diff
@@ -0,0 +1,24 @@
+window.onload = function () {
+    const tableElement = document.querySelector('.table');
+    const cellElements = tableElement.querySelectorAll('.cell')
+    const minusRowButton = tableElement.previousElementSibling;
+    const minusColumnButton = minusRowButton.previousElementSibling;
+    cellElements.forEach(function (item) {
+        item.addEventListener('mouseover', moveButtons);
```
Comment (kozubenko) [inline js/script.js:7]: onMouseOver

File: js/script.js [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+        setButtonsVisibility('visible');
+        minusColumnButton.style.marginLeft = event.target.offsetLeft - tableElement.offsetLeft + minusRowButton.offsetWidth + 'px';
+        minusRowButton.style.marginTop = event.target.offsetTop - tableElement.offsetTop + 'px';
+    }
+
+    function hideButtons() {
+        setButtonsVisibility('hidden');
+    }
+
+    function setButtonsVisibility(visibility) {
```
Comment (kozubenko) [inline js/script.js:21]: const buttonVisibility = ....
minusColumnButton.style.visibility = buttonVisibility === 'hidden' ? 'visible' : 'hidden'

File: js/script.js [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+    const minusRowButton = tableElement.previousElementSibling;
+    const minusColumnButton = minusRowButton.previousElementSibling;
+    cellElements.forEach(function (item) {
+        item.addEventListener('mouseover', moveButtons);
+    });
+    tableElement.addEventListener('mouseout', hideButtons);
+
+    function moveButtons(event) {
+        setButtonsVisibility('visible');
+        minusColumnButton.style.marginLeft = event.target.offsetLeft - tableElement.offsetLeft + minusRowButton.offsetWidth + 'px';
```
Comment (kozubenko) [inline js/script.js:13]: хуйня
читать - position: absolute, relative
left, top

File: js/script.js [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+        minusRowButton.style.marginTop = event.target.offsetTop - tableElement.offsetTop + 'px';
```
Comment (kozubenko) [inline js/script.js:14]: `${event.target.offsetTop - tableElement.offsetTop}px`

File: js/script.js [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+    const tableElement = document.querySelector('.table');
+    const cellElements = tableElement.querySelectorAll('.cell')
+    const minusRowButton = tableElement.previousElementSibling;
+    const minusColumnButton = minusRowButton.previousElementSibling;
+    cellElements.forEach(function (item) {
+        item.addEventListener('mouseover', moveButtons);
+    });
+    tableElement.addEventListener('mouseout', hideButtons);
+
+    function moveButtons(event) {
```
Comment (kozubenko) [inline js/script.js:11]: `function moveButtons({ target })`
`const { offsetLeft: a } = tableElement;`

or { currentTarget } - it should be better

File: js/script.js [hunk 17]
```diff
@@ -0,0 +1,24 @@
+window.onload = function () {
+    const tableElement = document.querySelector('.table');
+    const cellElements = tableElement.querySelectorAll('.cell')
+    const minusRowButton = tableElement.previousElementSibling;
```
Comment (kozubenko) [inline js/script.js:4]: change

File: js/script.js (hunk 12 above)
Comment (sergeymaksimenko) [reply js/script.js:7]: what do you mean?
item.addEventListener('**onmouseover**', moveButtons);
it doesn't work

File: js/script.js (hunk 13 above)
Comment (sergeymaksimenko) [reply js/script.js:21]: mb we refused that

**Pull request closed without merging**